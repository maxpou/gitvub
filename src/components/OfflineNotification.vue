<template>
  <div
    id="snackbar-offline"
    class="mdl-js-snackbar mdl-snackbar">
    <div class="mdl-snackbar__text" />
    <button
      class="mdl-snackbar__action"
      type="button"/>
  </div>
</template>
<script>
export default {
  props: {
    isOffline: {
      type: Boolean,
      default: false
    }
  },
  watch: {
    isOffline (offline) {
      if (!offline) {
        return
      }
      const snackbarContainer = document.querySelector('#snackbar-offline')
      snackbarContainer.MaterialSnackbar.showSnackbar({
        message: '⚠️ You are offline.',
        timeout: 5000
      })
    }
  }
}
</script>
