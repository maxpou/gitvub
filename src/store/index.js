import Vue from 'vue'
import Vuex from 'vuex'
import repositories from './modules/repositories'
import users from './modules/users'
import navigator from './modules/navigator'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

const store = new Vuex.Store({
  modules: {
    repositories,
    users,
    navigator
  },
  strict: debug
})

/* istanbul ignore if */
if (module.hot) {
  module.hot.accept([
    './modules/repositories',
    './modules/users'
  ], () => {
    const repositories = require('./modules/repositories').default
    store.hotUpdate({
      modules: {
        repositories,
        users,
        navigator
      }
    })
  })
}

export default store
