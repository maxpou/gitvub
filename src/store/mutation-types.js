export const RECEIVE_REPOSITORIES = 'RECEIVE_REPOSITORIES'
export const RECEIVE_LANGUAGES = 'RECEIVE_LANGUAGES'
export const FETCH_USER_REPOS = 'FETCH_USER_REPOS'
export const FETCH_REPO_DETAIL = 'FETCH_REPO_DETAIL'
export const RECEIVE_REPOSITORY_LANGUAGE = 'RECEIVE_REPOSITORY_LANGUAGE'
export const RECEIVE_REPOSITORY_README = 'RECEIVE_REPOSITORY_README'
