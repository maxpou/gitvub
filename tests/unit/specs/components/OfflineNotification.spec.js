import { mount } from '@vue/test-utils'
import OfflineNotification from '@/components/OfflineNotification.vue'
import store from '@/store'

describe('OfflineNotification.vue', () => {
  it('test initial rendering (when online)', () => {
    const wrapper = mount(OfflineNotification, {
      store,
      propsData: {
        isOffline: false
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('test initial rendering (when offline)', () => {
    const wrapper = mount(OfflineNotification, {
      store,
      propsData: {
        isOffline: true
      }
    })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
