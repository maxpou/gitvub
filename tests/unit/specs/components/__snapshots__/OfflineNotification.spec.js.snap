// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineNotification.vue test initial rendering (when offline) 1`] = `"<div id=\\"snackbar-offline\\" class=\\"mdl-js-snackbar mdl-snackbar\\"><div class=\\"mdl-snackbar__text\\"></div> <button type=\\"button\\" class=\\"mdl-snackbar__action\\"></button></div>"`;

exports[`OfflineNotification.vue test initial rendering (when online) 1`] = `"<div id=\\"snackbar-offline\\" class=\\"mdl-js-snackbar mdl-snackbar\\"><div class=\\"mdl-snackbar__text\\"></div> <button type=\\"button\\" class=\\"mdl-snackbar__action\\"></button></div>"`;
